lib/chef/resource/windows_certificate.rb

```diff
@@ -331,7 +380,44 @@ def acl_script(hash)
         # @raise [OpenSSL::PKCS12::PKCS12Error] When incorrect password is provided for PFX certificate
         #
         def fetch_cert_object(ext)
-          contents = ::File.binread(new_resource.source)
+          type = url_or_file?(new_resource.source)
+
+          if type == "file"
+            begin
+              ::File.exist?(new_resource.source)
+              contents = ::File.binread(new_resource.source)
+            rescue => exception
+              message = "Unable to load the certificate object from the specified local path : #{new_resource.source}\n"
+              message << exception.message
+              raise Chef::Exceptions::FileNotFound, message
+            end
```

Comment:
this should probably be `fetch_cert_object_from_file` along with `fetch_cert_object_from_url` below in order to break up `fetch_cert_object` a bit better.  the exception handling here is also a good sign that you need a new method.

Reply:
done!